### PR TITLE
feat: add command for previewing deployments

### PR DESCRIPTION
**Pull request by HomelessDinosaur**
Relies on: https://github.com/nitrictech/nitric/pull/543

### CODE REVIEW

File: cmd/stack.go [hunk 1]
```diff
@@ -176,6 +177,44 @@ nitric stack down -s aws -y`,
 	Args: cobra.ExactArgs(0),
 }
 
+var stackPreviewCommand = &cobra.Command{
+	Use:     "preview [-s stack]",
+	Short:   "Preview the deployment of a stack",
+	Long:    `Preview the deployment of a stack`,
+	Example: `nitric stack down -s aws`,
```
Comment (davemooreuws) [inline cmd/stack.go]:
```suggestion
	Example: `nitric stack preview -s aws`,
```

File: cmd/stack.go [hunk 2]
```diff
@@ -176,6 +177,44 @@ nitric stack down -s aws -y`,
...
+	Run: func(cmd *cobra.Command, args []string) {
+		s, err := stack.ConfigFromOptions()
+
+		if err != nil && strings.Contains(err.Error(), "No nitric stacks found") {
```
Comment (davemooreuws) [inline cmd/stack.go]: This block doesn't work for me. Yes sends the error "No nitric stacks found".
Same thing happens to the up command. Could be an unrelated issue to this PR

File: cmd/stack.go [hunk 3]
```diff
@@ -185,6 +224,11 @@ func init() {
 	stackUpdateCmd.Flags().BoolVarP(&forceStack, "force", "f", false, "force override previous deployment")
 	utils.CheckErr(stack.AddOptions(stackUpdateCmd, false))
 
+	stackCmd.AddCommand(command.AddDependencyCheck(stackPreviewCommand, command.Pulumi, command.Docker))
+	stackPreviewCommand.Flags().StringVarP(&envFile, "env-file", "e", "", "--env-file config/.my-env")
+	stackPreviewCommand.Flags().BoolVarP(&forceStack, "force", "f", false, "force override previous deployment")
```
Comment (davemooreuws) [inline cmd/stack.go]: Does preview need a force?

**Review by davemooreuws — CHANGES_REQUESTED** [review]

File: cmd/stack.go (hunk 3 above)
Comment (HomelessDinosaur) [reply cmd/stack.go]: If its in a conflicting state then it needs a force

File: cmd/stack.go (hunk 2 above)
Comment (HomelessDinosaur) [reply cmd/stack.go]: This was removed from `up` as well when it was refactored, there are a few lines commented out. Will update now.

**Review by davemooreuws — DISMISSED** [review]

**Comment by HomelessDinosaur** [comment]
Marking as draft until the Nitric provider PR gets merged.

**Pull request closed without merging**